src/components/Documents/DocumentTable.jsx

```diff
@@ -32,47 +32,166 @@ import { EmptyListNotification, LoadingAnimation } from '../Notification';
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center'
+    },
+    {
+      headerName: 'Expiration Date',
+      field: 'expiration date',
+      minWidth: 120,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center'
+    },
+    {
+      headerName: 'Preview File',
+      field: 'preview file',
+      minWidth: 100,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      sortable: false,
+      filterable: false,
+      renderCell: (fileUrl) => (
+        <GridActionsCellItem
+          key={String(fileUrl)}
+          icon={<FileOpenIcon />}
+          onClick={() => handleShowDocumentLocal(fileUrl)}
+          label="Preview"
+        />
+      )
+    },
+    {
+      headerName: 'Sharing',
+      field: 'sharing',
+      type: 'actions',
+      minWidth: 80,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      getActions: (data) => [
+        <GridActionsCellItem
+          icon={<ShareIcon />}
+          onClick={() => handleAclPermissionsModal('documents', data.name, data.type)}
+          label="Share"
+        />
+      ]
+    },
+    {
+      headerName: 'Delete',
+      field: 'actions',
+      type: 'actions',
+      minWidth: 80,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      getActions: (data) => [
+        <GridActionsCellItem
+          icon={<DeleteOutlineOutlinedIcon />}
+          onClick={() => handleSelectDeleteDoc(data.row.delete)}
+          label="Delete"
+        />
+      ]
+    }
   ];
 
   const determineDocumentsTable = documentListObject?.docList?.length ? (
     // render if documents
-    <ThemeProvider theme={theme}>
-      <Container>
-        <TableContainer component={Paper} sx={{ margin: '1rem 0' }}>
-          <Table aria-label="Documents Table">
-            <TableHead>
-              <TableRow>
-                {columnTitlesArray.map((columnTitle) => (
-                  <TableCell key={columnTitle} align="center">
-                    {columnTitle}
-                  </TableCell>
-                ))}
-              </TableRow>
-            </TableHead>
-            <TableBody>
-              {documentListObject?.docList.map((document) => (
-                <DocumentTableRow
-                  key={document.name}
-                  document={document}
-                  handleAclPermissionsModal={handleAclPermissionsModal}
-                  handleSelectDeleteDoc={handleSelectDeleteDoc}
-                />
-              ))}
-            </TableBody>
-          </Table>
-        </TableContainer>
-      </Container>
-    </ThemeProvider>
+    <Container>
+      <DataGrid
+        columns={columnTitlesArray}
+        rows={documentListObject?.docList.map((document) => ({
+          id: document.name,
+          type: document.type,
+          name: document.name,
+          description: document.description,
+          delete: document,
+          'upload date': document?.uploadDate.toLocaleDateString(),
```

Comment:
@leekahung let me know if this is the format you wanted.

Reply:
Yep, that would be good for `upload date`.

For `expiration date` since it could potentially be null from the user not including one, think we could have it leave an empty string if no expiration date was given. So something like:

```
'expiration date': document?.endDate?.toLocaleDateString() || '',
```

Or something that gives a similar result.

Reply:
Looks like `'expiration date': document?.endDate?.toLocaleDateString()` works too

Reply:
Ok, I'll add a test case for that.